api/uniter/unit_test.go

```diff
@@ -239,6 +239,67 @@ func (s *unitSuite) TestWatch(c *gc.C) {
 	wc.AssertOneChange()
 }
 
+func (s *unitSuite) TestWatchRelations(c *gc.C) {
+	w, err := s.apiUnit.WatchRelations()
+	c.Assert(err, jc.ErrorIsNil)
+	wc := watchertest.NewStringsWatcherC(c, w, s.BackingState.StartSync)
+	defer wc.AssertStops()
+
+	// Initial event.
+	wc.AssertChange()
+	wc.AssertNoChange()
+
+	// Change something other than the lifecycle and make sure it's
+	// not detected.
+	err = s.wordpressApplication.SetExposed()
+	c.Assert(err, jc.ErrorIsNil)
+	wc.AssertNoChange()
+
+	// Add another application and relate it to wordpress,
+	// check it's detected.
+	s.addMachineAppCharmAndUnit(c, "mysql")
+	rel := s.addRelation(c, "wordpress", "mysql")
+	wc.AssertChange(rel.String())
+
+	// Destroy the relation and check it's detected.
+	err = rel.Destroy()
+	c.Assert(err, jc.ErrorIsNil)
+	wc.AssertChange(rel.String())
+	wc.AssertNoChange()
+}
+
+func (s *unitSuite) TestSubordinateWatchRelations(c *gc.C) {
+	// A subordinate unit deployed with this wordpress unit shouldn't
+	// be notified about changes to logging mysql.
+	loggingRel, _, loggingUnit := s.addRelatedApplication(c, "wordpress", "logging", s.wordpressUnit)
+	password, err := utils.RandomPassword()
+	c.Assert(err, jc.ErrorIsNil)
+	err = loggingUnit.SetPassword(password)
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Add another principal app that we can relate logging to.
+	s.addMachineAppCharmAndUnit(c, "mysql")
+
+	api := s.OpenAPIAs(c, loggingUnit.Tag(), password)
+	uniter, err := api.Uniter()
+	c.Assert(err, jc.ErrorIsNil)
+	apiUnit, err := uniter.Unit(loggingUnit.Tag().(names.UnitTag))
+	c.Assert(err, jc.ErrorIsNil)
+
+	w, err := apiUnit.WatchRelations()
+	c.Assert(err, jc.ErrorIsNil)
+
+	wc := watchertest.NewStringsWatcherC(c, w, s.BackingState.StartSync)
+	defer wc.AssertStops()
+
+	wc.AssertChange(loggingRel.Tag().Id())
+	wc.AssertNoChange()
+
+	// Adding a subordinate relation to another application doesn't notify this unit.
```

Comment:
Can we just also add a subordinate relation that does trigger the watcher? I know it's elsewhere too, but to see the watcher trigger right next to the watcher not triggering is useful